assets/src/components/with-cropped-featured-image.js

```diff
@@ -0,0 +1,264 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { dispatch } from '@wordpress/data';
+
+/**
+ * External dependencies
+ */
+import { now } from 'lodash';
+
+const EXPECTED_WIDTH = 696;
+const EXPECTED_HEIGHT = 928;
+
+/**
+ * Gets a wrapped version of MediaUpload.
+ *
+ * Only applies to the MediaUpload in the Featured Image component, PostFeaturedImage.
+ * Suggests cropping of the featured image if it's not 696 x 928.
+ * Mostly copied from customize-controls.js.
+ *
+ * @param {Function} InitialMediaUpload The MediaUpload component, passed from the filter.
+ * @return {Function} The wrapped component.
+ */
+export default ( InitialMediaUpload ) => {
+	/**
+	 * Mostly copied from customize-controls.js, with slight changes.
+	 *
+	 * @see wp.media.HeaderControl
+	 */
+	return class FeaturedImageMediaUpload extends InitialMediaUpload {
+		/**
+		 * Constructs the class.
+		 */
+		constructor() {
+			super( ...arguments );
+
+			// This class should only be present in the MediaUpload for the Featured Image.
+			if ( this.props.modalClass && 'editor-post-featured-image__media-modal' === this.props.modalClass ) {
```

Comment:
It's not ideal to [use the modalClass](https://github.com/WordPress/gutenberg/blob/94f54892c214ebd014ec126a9575b7e902198ad9/packages/editor/src/components/post-featured-image/index.js#L50) to ensure that this `MediaUpload` is from the `PostFeaturedImage` component.

I couldn't find a better way to do this, though maybe there is one.

One alternative would be to filter the [PostFeaturedImage](https://github.com/WordPress/gutenberg/blob/94f54892c214ebd014ec126a9575b7e902198ad9/packages/editor/src/components/post-featured-image/index.js#L132) component. But that would probably violate DRY really badly.

Reply:
What about filtering `PostFeaturedImage` and only within that callback add the `MediaUpload` filter.

The `PostFeaturedImage` filter callback itself would just register that `MediaUpload` filter and return the original, unchanged component.

At least in PHP land that is possible. Not sure about the editor though.

But the `modalClass` check isn't too bad really, we could probably just keep it this way.

Reply:
OK, thanks